test/unit/query/search.js

```diff
@@ -270,6 +270,80 @@ module.exports.tests.query = function(test, common) {
     t.deepEqual(compiled.body, expected, 'valid search query with category filtering');
     t.end();
   });
+
+};
+
+module.exports.tests.city_state = function(test, common) {
+  test('only city and state set should return query', function(t) {
+    var clean = {
+      parsed_text: {
+        'city': 'city value',
+        'state': 'state value'
+      }
+    };
+
+    var query = generate(clean);
+
+    t.notEqual(query, undefined, 'should not have returned undefined');
+    t.end();
+
+  });
+
+  test('only city, state, and country set should return query', function(t) {
+    var clean = {
+      parsed_text: {
+        'city': 'city value',
+        'state': 'state value',
+        'country': 'country value'
+      }
+    };
+
+    var query = generate(clean);
+
+    t.notEqual(query, undefined, 'should not have returned undefined');
+    t.end();
+
+  });
+
+  test('city- OR state-only should return undefined', function(t) {
+    ['city', 'state'].forEach(function(placeType) {
+      var clean = {
+        parsed_text: {}
+      };
+
+      clean.parsed_text[placeType] = placeType + ' value';
+
+      var query = generate(clean);
+
+      t.equals(query, undefined, 'should have returned undefined');
+
+    });
+
+    t.end();
+
+  });
+
+  test('city and state with at least one other input: field should return undefined', function(t) {
+    ['query', 'category', 'number', 'neighbourhood', 'borough', 'postalcode', 'county'].forEach(function(placeType) {
```

Comment:
this is not so bad right now but we've seen for loops in unit tests get out of control before. Imagine what 6 months of changes quickly bolted on to this test could do. it does feel repetitive but having this split out into several tests is probably better.